### ISSUE TITLE
Useful endpoint request: recent/gameIds/by/{queue}

### ISSUE THREAD

**Issue by ruvane3** [RiotGames/developer-relations#127]
This new endpoint would make processing all games in a certain queue possible. Instead of doing spidering from players to build a sample of games that happened in a certain queue, this would directly tell you what gameIds happened. It would be much more concise, require much less client storage, much fewer api calls, and actually 100% cover the games in a queue.

### Simple Option:
**get/recent/gameIds/by/{queue}**
#### Parameters
-   queue (required)
#### Returns 

-   1000 most recent gameIds in that queue

### More Robust Option:
**get/gameIds**

#### Parameters

- begin timestamp (required)
-  end timestamp (optional - default to current time)
-  begin index (optional - for pagination)
-  end index (optional - for pagination)
-  queue (optional - default to all queues)
-  average tier of players (optional - optional to implement as well depending on difficulty)

#### Returns

-  1000 gameIds (paginated by indexes and timestamp)
-  total number of gameIds between begin time and end time
-  start index
-  end index

**Comment by stelar7** [comment]
Whats the need for this? you get the same data from the matchlist, its just limited to 100(?) instead of 1000...

**Comment by ruvane3** [comment]
> Whats the need for this?
> Most people dont want the most recent games, they just want their own games, and for that you get the same data from the matchlist..

You are right that if you only care about a certain player's games then the matchlist endpoint works really well. This is for analysis - especially analysis that requires looking at all games (not a sample).

My use case needs to look at all games that happen - not just a sample:
![sneaky_darshan](https://user-images.githubusercontent.com/9937507/57822406-65c0de00-7748-11e9-9d06-3ea05de35459.PNG)

Basically it is building a network of who beat who in ranked and how in maybe just 4 games you are better than bjergsen by the transitive property.  The problem is I need to look at all of the games - not just a sample.


Although this would be useful in general too:

- winrate of champion week by week (day by day? hour by hour?) becomes much easier and more accurate (spidering + sample vs just looking at all of the games)
- build cool stats (you played the game with the most assists in NA ranked for a particular day, your game was one of 12 this month where there were 3 penta kills by a player, etc...)
- I'm pretty sure all things that are related to general analysis would prefer this endpoint over spidering + sample

**Comment by RiotTuxedo** [comment]
We're looking into a couple possibilities related to exposing a list of active or recently completed matches. No ETA, but it's definitely on our radar.

**Comment by RiotTuxedo** [comment]
Someone asked for an update to this topic in #office-hours on [Discord](https://discord.gg/riotapi).

```Still looking into this, if there is a change it would likely happen along side TFT getting added to match history. I don't expect any new endpoints before then.```

**Comment by ruvane3** [comment]
Any ETA update on this? Is it still on the radar?

**Comment by MingweiSamuel** [comment]
Valorant now has a similar endpoint: https://twitter.com/RiotGamesDevRel/status/1305651691925061632
